src/client/src/rt-platforms/openfin-platform/snapshots/index.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  OPENFIN_SNAPSHOT_CURRENT,
+  OPENFIN_SNAPSHOT_DEFAULT_NAME,
+  OPENFIN_SNAPSHOT_NAMES,
+  OPENFIN_SNAPSHOTS,
+} from '../StorageItems'
+import canned from './canned.json'
+import { union } from 'lodash'
+
+export const resetCurrentSnapshotName = () => {
+  setCurrentSnapshotName(OPENFIN_SNAPSHOT_DEFAULT_NAME)
+}
+export const getCurrentSnapshotName = () => {
+  return window.localStorage.getItem(OPENFIN_SNAPSHOT_CURRENT) || OPENFIN_SNAPSHOT_DEFAULT_NAME
+}
+export const getSnapshotNames = () => {
+  const snapshotNamesStr = window.localStorage.getItem(OPENFIN_SNAPSHOT_NAMES)
+  if (snapshotNamesStr) {
+    return JSON.parse(snapshotNamesStr)
+  }
+  return []
+}
+export const getSnapshots = () => {
+  const snapshotsStr = window.localStorage.getItem(OPENFIN_SNAPSHOTS)
+  if (snapshotsStr) {
+    return JSON.parse(snapshotsStr)
+  }
+  return {}
+}
+
+const setCurrentSnapshotName = (snapshotName: string) => {
+  window.localStorage.setItem(OPENFIN_SNAPSHOT_CURRENT, snapshotName)
+}
+const setSnapshotNames = (snapshotNames: string[]) => {
+  window.localStorage.setItem(OPENFIN_SNAPSHOT_NAMES, JSON.stringify(snapshotNames))
+}
+const setSnapshots = (snapshots: any) => {
+  window.localStorage.setItem(OPENFIN_SNAPSHOTS, JSON.stringify(snapshots))
+}
+const setPlatformSnapshotName = async (platform: any, platformSnapshotName: string) => {
+  await platform.setContext({ platformSnapshotName })
+  return platformSnapshotName
+}
+
+export const applySnapshotFromStorage = (snapshotName: string) => {
+  return fin.Platform.getCurrent().then((platform: any) => {
+    const snapshotNames = getSnapshotNames()
+    const snapshots = getSnapshots()
+
+    if (snapshotNames.includes(snapshotName)) {
+      setCurrentSnapshotName(snapshotName)
+      return platform.applySnapshot(snapshots.snapshots[snapshotName], {
+        closeExistingWindows: true,
+      })
+    }
+    return false
+  })
+}
```

Comment:
This is very, very messy. It tries to create pre-canned window layout snapshots at startup, from json config files, and put them in isolated storage, and then display them in UI.

It doesn't work with this version of OpenFin anyway.